doc/hardware/emulator/bus_emulators.rst

```diff
@@ -0,0 +1,168 @@
+for an I2C compass such that it appears on the I2C bus and can be used
+just like a real hardware device.
+
+Emulators often implement special features for testing. For example a
+compass may support returning bogus data if the I2C bus speed is too
+high, or may return invalid measurements if calibration has not yet
+been completed. This allows for testing that high-level code can
+handle these situations correctly. Test coverage can therefore
+approach 100% if all failure conditions are emulated.
+
+Concept
+=======
+
+The diagram below shows application code / high-level tests at the top.
+This is the ultimate application we want to run.
+
+.. figure:: img/arch.svg
+   :align: center
+   :alt: Emulator architecture showing tests, emulators and drivers
+
+Below that are peripheral drivers, such as the AT24 EEPROM driver. We can test
+peripheral drivers using an emulation driver connected via a emulated I2C
+controller/emulator which passes I2C traffic from the AT24 driver to the AT24
+simulator.
+
+Separately we can test the STM32 and NXP I2C drivers on real hardware using API
+tests. These require some sort of device attached to the bus, but with this, we
+can validate much of the driver functionality.
+
+Putting the two together, we can test the application and peripheral code
+entirely on native_sim. Since we know that the I2C driver on the real hardware
+works, we should expect the application and peripheral drivers to work on the
+real hardware also.
+
+Using the above framework we can test an entire application (e.g. Embedded
+Controller) on native_sim using emulators for all non-chip drivers.
+
+With this approach we can:
+
+* Write individual tests for each driver (green), covering all failure modes,
+  error conditions, etc.
+
+* Ensure 100% test coverage for drivers (green)
+
+* Write tests for combinations of drivers, such as GPIOs provided by an I2C GPIO
+  expander driver talking over an I2C bus, with the GPIOs controlling a charger.
+  All of this can work in the emulated environment or on real hardware.
+
+* Write a complex application that ties together all of these pieces and runs on
+  native_sim. We can develop on a host, use source-level debugging, etc.
+
+* Transfer the application to any board which provides the required features
+  (e.g. I2C, enough GPIOs), by adding Kconfig and devicetree fragments.
+
+Creating a Device Driver Emulator
+=================================
+
+The emulator subsystem is modeled on the :ref:`device_model_api`.  You create
+an emulator instance using one of the :c:func:`EMUL_DT_DEFINE()` or
+:c:func:`EMUL_DT_INST_DEFINE()` APIs.
+
+Emulators for peripheral devices reuse the same devicetree node as the real
+device driver. This means that your emulator defines `DT_DRV_COMPAT` using the
+same ``compat`` value from the real driver.
+
+.. code-block:: C
+
+  /* From drivers/sensor/bm160/bm160.c */
+  #define DT_DRV_COMPAT bosch_bmi160
+
+  /* From subsys/emul/emul_bmi160.c */
```

Comment:
This file moved to live right next to the driver, it's in `drivers/sensor/bmi160/`

Reply:
I updated this path in this last push. (Note though I this PR was not really changing this part of the text)